src/cbapi/psc/threathunter/models.py

```diff
@@ -175,3 +177,694 @@ def children(self):
+            raise InvalidObjectError("missing Report ID")
+
+        if not self._from_watchlist:
+            raise InvalidObjectError("ignoring only applies to watchlist reports")
+
+        self._cb.put_object("/threathunter/watchlistmgr/v1/report/{}/ignore".format(self.id), None)
+
+    def unignore(self):
+        """Removes the ignore status on this report.
+
+        Only watchlist reports have an ignore status.
+
+        :raises InvalidObjectError: if `id` is missing or this report is not from a watchlist
+        """
+        if not self.id:
+            raise InvalidObjectError("missing Report ID")
+
+        if not self._from_watchlist:
+            raise InvalidObjectError("ignoring only applies to watchlist reports")
+
+        self._cb.delete_object("/threathunter/watchlistmgr/v1/report/{}/ignore".format(self.id))
+
+    @property
+    def custom_severity(self):
+        """Returns the custom severity for this report.
+
+        :return: The custom severity for this report, if it exists
+        :rtype: :py:class:`ReportSeverity`
+        :raise InvalidObjectError: if `id` is missing or this report is from a watchlist
+        """
+        if not self.id:
+            raise InvalidObjectError("missing report ID")
+        if self._from_watchlist:
+            raise InvalidObjectError("watchlist reports don't have custom severities")
+
+        resp = self._cb.get_object("/threathunter/watchlistmgr/v1/severity/report/{}".format(self.id))
+        return ReportSeverity(self._cb, initial_data=resp)
+
+    @custom_severity.setter
+    def custom_severity(self, sev_level):
+        """Sets or removed the custom severity for this report
+
+        :param int sev_level: the new severity, or None to remove the custom severity
+        :return: The new custom severity, or None if removed
+        :rtype: :py:class:`ReportSeverity` or None
+        :raise InvalidObjectError: if `id` is missing or this report is from a watchlist
+        """
+        if not self.id:
+            raise InvalidObjectError("missing report ID")
+        if self._from_watchlist:
+            raise InvalidObjectError("watchlist reports don't have custom severities")
+
+        url = "/threathunter/watchlistmgr/v1/severity/report/{}".format(self.id)
+
+        if sev_level is None:
+            self._cb.delete_object(url)
+            return
+
+        args = {
+            "report_id": self.id,
+            "severity": sev_level,
+        }
+
+        resp = self._cb.put_object(url, args).json()
+        return ReportSeverity(self._cb, initial_data=resp)
+
+    @property
+    def iocs_(self):
+        """Returns a list of :py:class:`IOC_V2` associated with this report.
+
+        >>> for ioc in report.iocs_:
+        ...     print(ioc.values)
+
+        :return: a list of IOCs
+        :rtype: list(:py:class:`IOC_V2`)
+        """
+        if not self.iocs_v2:
+            return []
+
+        # NOTE(ww): This name is underscored because something in the model
```

Comment:
any idea why this is happening?

Reply:
Not entirely -- I think it has to do with the interaction between `@property` and `NewBaseModel.__getattr__()` (which in turn calls `__getattribute__`), but I hadn't looked fully into it.

Reply:
OK got it... we can take a look later. We can do the underscores for now. Thanks